src/cunumeric/stat/histogram.cc

```diff
@@ -0,0 +1,183 @@
+/* Copyright 2023 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "cunumeric/stat/histogram.h"
+#include "cunumeric/stat/histogram_template.inl"
+
+#include "cunumeric/stat/histogram_cpu.h"
+#include "cunumeric/stat/histogram_impl.h"
+
+#include <algorithm>
+#include <numeric>
+#include <tuple>
+
+// #define _DEBUG
+#ifdef _DEBUG
+#include <iostream>
+#include <iterator>
+#endif
+
+namespace cunumeric {
+using namespace legate;
+
+namespace detail {
```

Comment:
First off, there's no such thing as "anonymous" namespace. The compiler gives them a (mangled) name, a different one for each TU where they're encountered. 

Second, this was a useful practice we had in Rapids, to nest a `details` namespace to suggest that we're dealing with implementation details, the most likely ones to change in case of having to fix bugs or enhance functionality. Anonymous namespaces serve a different purpose: reusing same names in different TUs (formerly achieved via `static` functions).

Reply:
I agree with that practice. I'm just a little worried that `get_accessor` is a pretty generic name, and someone else in some other file might also define a `get_accessor` template, causing a clash.

Reply:
Fixed according to the 4 points we agreed:

1. unify accessor helpers to one undifferentiated form for every ecosystem (GPU/CPU/OMP);
2. put them in a header (*_gen.h), under a deeper nested namespace, say, cunumeric::detail::accessor namespace;
3. move the copy inside make_accessor_copy();
4. do the copying via thrust::copy_n();